### PR TITLE
Fix type definition of curl "certinfo" array

### PR DESCRIPTION
**Pull request by cs278**
With 1.5.0 (#1072) I'm now getting `Offset 'Cert' on array<int, string> in isset() does not exist.` in my code: https://phpstan.org/r/4d09807c-a3f3-479e-a638-5baa14eba1ad

The array structure actually looks like this:

```
[
    0 => array:23 [
      "Subject" => "C = US, ST = California, L = Los Angeles, O = Internet\C2\A0Corporation\C2\A0for\C2\A0Assigned\C2\A0Names\C2\A0and\C2\A0Numbers, CN = www.example.org"
      "Issuer" => "C = US, O = DigiCert Inc, CN = DigiCert TLS RSA SHA256 2020 CA1"
      "Version" => "2"
      "Serial Number" => "0faa63109307bc3d414892640ccd4d9a"
      "Signature Algorithm" => "sha256WithRSAEncryption"
      "Public Key Algorithm" => "rsaEncryption"
      "X509v3 Authority Key Identifier" => "keyid:B7:6B:A2:EA:A8:AA:84:8C:79:EA:B4:DA:0F:98:B2:C5:95:76:B9:F4"
      "X509v3 Subject Key Identifier" => "F7:2A:09:D0:24:5B:11:71:EE:BA:BE:F4:3E:1C:3D:56:12:88:16:BB"
      "X509v3 Subject Alternative Name" => "DNS:www.example.org,DNS:example.net,DNS:example.edu,DNS:example.com,DNS:example.org,DNS:www.example.com,DNS:www.example.edu,DNS:www.example.net"
      "X509v3 Key Usage" => "DigitalSignature,KeyEncipherment"
      "X509v3 Extended Key Usage" => "TLSWebServerAuthentication,TLSWebClientAuthentication"
      "X509v3 CRL Distribution Points" => """
        , FullName:, URI:http://crl3.digicert.com/DigiCertTLSRSASHA2562020CA1-4.crl, \n
        FullName:, URI:http://crl4.digicert.com/DigiCertTLSRSASHA2562020CA1-4.crl
        """
      "X509v3 Certificate Policies" => "Policy:2.23.140.1.2.2, CPS:http://www.digicert.com/CPS"
      "Authority Information Access" => "OCSP-URI:http://ocsp.digicert.com, CAIssuers-URI:http://cacerts.digicert.com/DigiCertTLSRSASHA2562020CA1-1.crt"
      "X509v3 Basic Constraints" => "CA:FALSE"
      "CT Precertificate SCTs" => "SignedCertificateTimestamp:, Version:v1(0x0), LogID:E8:3E:D0:DA:3E:F5:06:35:32:E7:57:28:BC:89:6B:C9:, 03:D3:CB:D1:11:6B:EC:EB:69:E1:77:7D:6D:06:BD:6E, Timestamp:Mar1422:48:46.1972022GMT, Extensions:none, Signature:ecdsa-with-SHA256, 30:44:02:20:31:78:F3:53:EB:46:7F:BD:32:53:3D:5A:, F2:88:6C:D7:AC:BC:42:34:94:2A:B5:E5:0B:92:0B:81:, 10:C6:C6:FE:02:20:47:74:30:BC:82:C9:21:60:14:7E:, F4:9F:25:0F:F3:32:6A:AB:D8:90:DA:9D:61:3C:BB:36:, C2:E0:45:9B:73:00, SignedCertificateTimestamp:, Version:v1(0x0), LogID:35:CF:1"
      "Start date" => "Mar 14 00:00:00 2022 GMT"
      "Expire date" => "Mar 14 23:59:59 2023 GMT"
      "RSA Public Key" => "2048"
      "rsa(n)" => "955D96639AE51A7D5FA70BEE0618F49D505C3710B1907506FE9246E07BDA25E9FB60867598B73BFCFE2C9CE2598847F05854E290735B71409C622286F0990579B64E60C73553D3E212D556194E24B98BB1897A43F2830F8A7D41D1282B17B80A7CED7FFC8DB0789740523819B970423C0961CF61271DD7735D4B709BD5EE38B3BF63A2C04D532C6E9F3F9579DFA3F6D7D14809EC2D1A56A3AED3B03456BE8E90FC705A63AF1777112A154233B538079D4A2BE22C5912E6F1C556D486978F0846CD6EC147A108DECC424385A773A1D53D2826A262D65969081853C0C1C594078342D76CA108EE625BF77134C58D2CC455B67640CAAE85A59F3014AEA3E0E33033"
      "rsa(e)" => "10001"
      "Signature" => "aa:9f:be:5d:91:1b:ad:e4:4e:4e:cc:8f:07:64:44:35:b4:ad:3b:13:3f:c1:29:d8:b4:ab:f3:42:51:49:46:3b:d6:cf:1e:41:83:e1:0b:57:2f:83:69:79:65:07:6f:59:03:8c:51:94:89:18:10:3e:1e:5c:ed:ba:3d:8e:4f:1a:14:92:d3:2b:ff:d4:98:cb:a7:93:0e:bc:b7:1b:93:a4:42:42:46:d9:e5:b1:1a:6b:68:2a:9b:2e:48:a9:2f:1d:2a:b0:e3:f8:20:94:54:81:50:2e:ee:d7:e0:20:7a:7b:2e:67:fb:fa:d8:17:a4:5b:dc:ca:00:62:ef:23:af:7a:58:f0:7a:74:0c:bd:4d:43:f1:8c:02:87:dc:e3:ae:09:d2:f7:fa:37:3c:d2:4b:ab:04:e5:43:a5:d2:55:11:0e:41:87:5f:38:a8:e5:7a:5e:4c:46:b8:b6:fa:3f:c3:4b:cd:40:35:ff:e0:a4:71:74:0a:c1:20:8b:e3:54:47:84:d5:18:bd:51:9b:40:5d:dd:42:30:12:d1:3a:a5:63:9a:af:90:08:d6:1b:d1:71:0b:06:71:90:eb:ae:ad:af:ba:5f:c7:db:6b:1e:78:a2:b4:d1:06:23:a7:63:f3:b5:43:fa:56:8c:50:17:7b:1c:1b:4e:10:6b:22:0e:84:52:94:"
      "Cert" => """
        -----BEGIN CERTIFICATE-----\n
        MIIHRzCCBi+gAwIBAgIQD6pjEJMHvD1BSJJkDM1NmjANBgkqhkiG9w0BAQsFADBP\n
        MQswCQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMSkwJwYDVQQDEyBE\n
        aWdpQ2VydCBUTFMgUlNBIFNIQTI1NiAyMDIwIENBMTAeFw0yMjAzMTQwMDAwMDBa\n
        Fw0yMzAzMTQyMzU5NTlaMIGWMQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZv\n
        cm5pYTEUMBIGA1UEBxMLTG9zIEFuZ2VsZXMxQjBABgNVBAoMOUludGVybmV0wqBD\n
        b3Jwb3JhdGlvbsKgZm9ywqBBc3NpZ25lZMKgTmFtZXPCoGFuZMKgTnVtYmVyczEY\n
        MBYGA1UEAxMPd3d3LmV4YW1wbGUub3JnMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8A\n
        MIIBCgKCAQEAlV2WY5rlGn1fpwvuBhj0nVBcNxCxkHUG/pJG4HvaJen7YIZ1mLc7\n
        /P4snOJZiEfwWFTikHNbcUCcYiKG8JkFebZOYMc1U9PiEtVWGU4kuYuxiXpD8oMP\n
        in1B0SgrF7gKfO1//I2weJdAUjgZuXBCPAlhz2EnHddzXUtwm9XuOLO/Y6LATVMs\n
        bp8/lXnfo/bX0UgJ7C0aVqOu07A0Vr6OkPxwWmOvF3cRKhVCM7U4B51KK+IsWRLm\n
        8cVW1IaXjwhGzW7BR6EI3sxCQ4Wnc6HVPSgmomLWWWkIGFPAwcWUB4NC12yhCO5i\n
        W/dxNMWNLMRVtnZAyq6FpZ8wFK6j4OMwMwIDAQABo4ID1TCCA9EwHwYDVR0jBBgw\n
        FoAUt2ui6qiqhIx56rTaD5iyxZV2ufQwHQYDVR0OBBYEFPcqCdAkWxFx7rq+9D4c\n
        PVYSiBa7MIGBBgNVHREEejB4gg93d3cuZXhhbXBsZS5vcmeCC2V4YW1wbGUubmV0\n
        ggtleGFtcGxlLmVkdYILZXhhbXBsZS5jb22CC2V4YW1wbGUub3Jngg93d3cuZXhh\n
        bXBsZS5jb22CD3d3dy5leGFtcGxlLmVkdYIPd3d3LmV4YW1wbGUubmV0MA4GA1Ud\n
        DwEB/wQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwgY8GA1Ud\n
        HwSBhzCBhDBAoD6gPIY6aHR0cDovL2NybDMuZGlnaWNlcnQuY29tL0RpZ2lDZXJ0\n
        VExTUlNBU0hBMjU2MjAyMENBMS00LmNybDBAoD6gPIY6aHR0cDovL2NybDQuZGln\n
        aWNlcnQuY29tL0RpZ2lDZXJ0VExTUlNBU0hBMjU2MjAyMENBMS00LmNybDA+BgNV\n
        HSAENzA1MDMGBmeBDAECAjApMCcGCCsGAQUFBwIBFhtodHRwOi8vd3d3LmRpZ2lj\n
        ZXJ0LmNvbS9DUFMwfwYIKwYBBQUHAQEEczBxMCQGCCsGAQUFBzABhhhodHRwOi8v\n
        b2NzcC5kaWdpY2VydC5jb20wSQYIKwYBBQUHMAKGPWh0dHA6Ly9jYWNlcnRzLmRp\n
        Z2ljZXJ0LmNvbS9EaWdpQ2VydFRMU1JTQVNIQTI1NjIwMjBDQTEtMS5jcnQwCQYD\n
        VR0TBAIwADCCAXwGCisGAQQB1nkCBAIEggFsBIIBaAFmAHUA6D7Q2j71BjUy51co\n
        vIlryQPTy9ERa+zraeF3fW0GvW4AAAF/ip6hdQAABAMARjBEAiAxePNT60Z/vTJT\n
        PVryiGzXrLxCNJQqteULkguBEMbG/gIgR3QwvILJIWAUfvSfJQ/zMmqr2JDanWE8\n
        uzbC4EWbcwAAdQA1zxkbv7FsV78PrUxtQsu7ticgJlHqP+Eq76gDwzvWTAAAAX+K\n
        nqF8AAAEAwBGMEQCIDspTxwkUBpEoeA+IolNYwOKl9Yxmwk816yd0O2IJPZcAiAV\n
        8TWhoOLiiqGKnY02CdcGXOzAzC7tT6m7OtLAku2+WAB2ALNzdwfhhFD4Y4bWBanc\n
        EQlKeS2xZwwLh9zwAw55NqWaAAABf4qeoYcAAAQDAEcwRQIgKR7qwPLQb6UT2+S7\n
        w7uQsbsDZfZVX/g8FkBtAltaTpACIQDLdtedRNGNhuzYpB6gmBBydhtSQi5YZLsp\n
        FvaVHpeW1zANBgkqhkiG9w0BAQsFAAOCAQEAqp++XZEbreROTsyPB2RENbStOxM/\n
        wSnYtKvzQlFJRjvWzx5Bg+ELVy+DaXllB29ZA4xRlIkYED4eXO26PY5PGhSS0yv/\n
        1JjLp5MOvLcbk6RCQkbZ5bEaa2gqmy5IqS8dKrDj+CCUVIFQLu7X4CB6ey5n+/rY\n
        F6Rb3MoAYu8jr3pY8Hp0DL1NQ/GMAofc464J0vf6NzzSS6sE5UOl0lURDkGHXzio\n
        5XpeTEa4tvo/w0vNQDX/4KRxdArBIIvjVEeE1Ri9UZtAXd1CMBLROqVjmq+QCNYb\n
        0XELBnGQ666tr7pfx9trHniitNEGI6dj87VD+laMUBd7HBtOEGsiDoRSlA==\n
        -----END CERTIFICATE-----\n
        """
    ]
    1 => array:21 [
      "Subject" => "C = US, O = DigiCert Inc, CN = DigiCert TLS RSA SHA256 2020 CA1"
      "Issuer" => "C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert Global Root CA"
      "Version" => "2"
      "Serial Number" => "06d8d904d5584346f68a2fa754227ec4"
      "Signature Algorithm" => "sha256WithRSAEncryption"
      "Public Key Algorithm" => "rsaEncryption"
      "X509v3 Basic Constraints" => "CA:TRUE,pathlen:0"
      "X509v3 Subject Key Identifier" => "B7:6B:A2:EA:A8:AA:84:8C:79:EA:B4:DA:0F:98:B2:C5:95:76:B9:F4"
      "X509v3 Authority Key Identifier" => "keyid:03:DE:50:35:56:D1:4C:BB:66:F0:A3:E2:1B:1B:C3:97:B2:3D:D1:55"
      "X509v3 Key Usage" => "DigitalSignature,CertificateSign,CRLSign"
      "X509v3 Extended Key Usage" => "TLSWebServerAuthentication,TLSWebClientAuthentication"
      "Authority Information Access" => "OCSP-URI:http://ocsp.digicert.com, CAIssuers-URI:http://cacerts.digicert.com/DigiCertGlobalRootCA.crt"
      "X509v3 CRL Distribution Points" => ", FullName:, URI:http://crl3.digicert.com/DigiCertGlobalRootCA.crl"
      "X509v3 Certificate Policies" => "Policy:2.16.840.1.114412.2.1, Policy:2.23.140.1.1, Policy:2.23.140.1.2.1, Policy:2.23.140.1.2.2, Policy:2.23.140.1.2.3"
      "Start date" => "Apr 14 00:00:00 2021 GMT"
      "Expire date" => "Apr 13 23:59:59 2031 GMT"
      "RSA Public Key" => "2048"
      "rsa(n)" => "C14BB3654770BCDD4F58DBEC9CEDC366E51F311354AD4A66461F2C0AEC6407E52EDCDCB90A20EDDFE3C4D09E9AA97A1D8288E51156DB1E9F58C251E72C340D2ED292E156CBF1795FB3BB87CA25037B9A52416610604F571349F0E8376783DFE7D34B674C2251A6DF0E9910ED57517426E27DC7CA622E131B7F238825536FC13458008B84FFF8BEA75849227B96ADA2889B15BCA07CDFE951A8D5B0ED37E236B4824B62B5499AECC767D6E33EF5E3D6125E44F1BF71427D58840380B18101FAF9CA32BBB48E278727C52B74D4A8D697DEC364F9CACE53A256BC78178E490329AEFB494FA415B9CEF25C19576D6B79A72BA2272013B5D03D40D321300793EA99F5"
      "rsa(e)" => "10001"
      "Signature" => "80:32:ce:5e:0b:dd:6e:5a:0d:0a:af:e1:d6:84:cb:c0:8e:fa:85:70:ed:da:5d:b3:0c:f7:2b:75:40:fe:85:0a:fa:f3:31:78:b7:70:4b:1a:89:58:ba:80:bd:f3:6b:1d:e9:7e:cf:0b:ba:58:9c:59:d4:90:d3:fd:6c:fd:d0:98:6d:b7:71:82:5b:cf:6d:0b:5a:09:d0:7b:de:c4:43:d8:2a:a4:de:9e:41:26:5f:bb:8f:99:cb:dd:ae:e1:a8:6f:9f:87:fe:74:b7:1f:1b:20:ab:b1:4f:c6:f5:67:5d:5d:9b:3c:e9:ff:69:f7:61:6c:d6:d9:f3:fd:36:c6:ab:03:88:76:d2:4b:2e:75:86:e3:fc:d8:55:7d:26:c2:11:77:df:3e:02:b6:7c:f3:ab:7b:7a:86:36:6f:b8:f7:d8:93:71:cf:86:df:73:30:fa:7b:ab:ed:2a:59:c8:42:84:3b:11:17:1a:52:f3:c9:0e:14:7d:a2:5b:72:67:ba:71:ed:57:47:66:c5:b8:02:4a:65:34:5e:8b:d0:2a:3c:20:9c:51:99:4c:e7:52:9e:f7:6b:11:2b:0d:92:7e:1d:e8:8a:eb:36:16:43:87:ea:2a:63:bf:75:3f:eb:de:c4:03:bb:0a:3c:f7:30:ef:eb:af:4c:fc:8b:36:10:73:3e:f3:a4:"
      "Cert" => """
        -----BEGIN CERTIFICATE-----\n
        MIIEvjCCA6agAwIBAgIQBtjZBNVYQ0b2ii+nVCJ+xDANBgkqhkiG9w0BAQsFADBh\n
        MQswCQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMRkwFwYDVQQLExB3\n
        d3cuZGlnaWNlcnQuY29tMSAwHgYDVQQDExdEaWdpQ2VydCBHbG9iYWwgUm9vdCBD\n
        QTAeFw0yMTA0MTQwMDAwMDBaFw0zMTA0MTMyMzU5NTlaME8xCzAJBgNVBAYTAlVT\n
        MRUwEwYDVQQKEwxEaWdpQ2VydCBJbmMxKTAnBgNVBAMTIERpZ2lDZXJ0IFRMUyBS\n
        U0EgU0hBMjU2IDIwMjAgQ0ExMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKC\n
        AQEAwUuzZUdwvN1PWNvsnO3DZuUfMRNUrUpmRh8sCuxkB+Uu3Ny5CiDt3+PE0J6a\n
        qXodgojlEVbbHp9YwlHnLDQNLtKS4VbL8Xlfs7uHyiUDe5pSQWYQYE9XE0nw6Ddn\n
        g9/n00tnTCJRpt8OmRDtV1F0JuJ9x8piLhMbfyOIJVNvwTRYAIuE//i+p1hJInuW\n
        raKImxW8oHzf6VGo1bDtN+I2tIJLYrVJmuzHZ9bjPvXj1hJeRPG/cUJ9WIQDgLGB\n
        Afr5yjK7tI4nhyfFK3TUqNaX3sNk+crOU6JWvHgXjkkDKa77SU+kFbnO8lwZV21r\n
        eacroicgE7XQPUDTITAHk+qZ9QIDAQABo4IBgjCCAX4wEgYDVR0TAQH/BAgwBgEB\n
        /wIBADAdBgNVHQ4EFgQUt2ui6qiqhIx56rTaD5iyxZV2ufQwHwYDVR0jBBgwFoAU\n
        A95QNVbRTLtm8KPiGxvDl7I90VUwDgYDVR0PAQH/BAQDAgGGMB0GA1UdJQQWMBQG\n
        CCsGAQUFBwMBBggrBgEFBQcDAjB2BggrBgEFBQcBAQRqMGgwJAYIKwYBBQUHMAGG\n
        GGh0dHA6Ly9vY3NwLmRpZ2ljZXJ0LmNvbTBABggrBgEFBQcwAoY0aHR0cDovL2Nh\n
        Y2VydHMuZGlnaWNlcnQuY29tL0RpZ2lDZXJ0R2xvYmFsUm9vdENBLmNydDBCBgNV\n
        HR8EOzA5MDegNaAzhjFodHRwOi8vY3JsMy5kaWdpY2VydC5jb20vRGlnaUNlcnRH\n
        bG9iYWxSb290Q0EuY3JsMD0GA1UdIAQ2MDQwCwYJYIZIAYb9bAIBMAcGBWeBDAEB\n
        MAgGBmeBDAECATAIBgZngQwBAgIwCAYGZ4EMAQIDMA0GCSqGSIb3DQEBCwUAA4IB\n
        AQCAMs5eC91uWg0Kr+HWhMvAjvqFcO3aXbMM9yt1QP6FCvrzMXi3cEsaiVi6gL3z\n
        ax3pfs8LulicWdSQ0/1s/dCYbbdxglvPbQtaCdB73sRD2Cqk3p5BJl+7j5nL3a7h\n
        qG+fh/50tx8bIKuxT8b1Z11dmzzp/2n3YWzW2fP9NsarA4h20ksudYbj/NhVfSbC\n
        EXffPgK2fPOre3qGNm+499iTcc+G33Mw+nur7SpZyEKEOxEXGlLzyQ4UfaJbcme6\n
        ce1XR2bFuAJKZTRei9AqPCCcUZlM51Ke92sRKw2Sfh3oius2FkOH6ipjv3U/697E\n
        A7sKPPcw7+uvTPyLNhBzPvOk\n
        -----END CERTIFICATE-----\n
        """
    ]
]
```

### CODE REVIEW

**Comment by staabm** [comment]
@PrinsFrank what do you think about this change?

File: tests/PHPStan/Analyser/data/curl_getinfo.php [hunk 1]
```diff
@@ -14,10 +14,10 @@ public function bar()
...
+		assertType('(array{url: string, content_type: string|null, http_code: int, header_size: int, request_size: int, filetime: int, ssl_verify_result: int, redirect_count: int, total_time: float, namelookup_time: float, connect_time: float, pretransfer_time: float, size_upload: float, size_download: float, speed_download: float, speed_upload: float, download_content_length: float, upload_content_length: float, starttransfer_time: float, redirect_time: float, redirect_url: string, primary_ip: string, certinfo: array<int, array<string, string>>, primary_port: int, local_ip: string, local_port: int, http_version: int, protocol: int, ssl_verifyresult: int, scheme: string}|false)', curl_getinfo($handle, PHP_INT_MAX));
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/curl_getinfo.php:17]: Do we need the same change in https://github.com/phpstan/phpstan-src/blob/cb590eee593f69bab554fdb0e374f892ce7e8eb2/tests/PHPStan/Analyser/data/curl_getinfo_7.3.php ?

**Comment by PrinsFrank** [comment]
> @PrinsFrank what do you think about this change?

Looks good! My intention was to document all the keys in the certinfo in the original PR, but there are many options and I couldn't find an exhaustive list of them all so I apparently documented them wrongly. ;)

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**